Rewrite validators using schemas

leplatrem opened the pull request:
- [x] Rewrite validators using reusable schemas (required by #230)
- [x] Move tests of validators out of test_views.py (separation of concerns : test_views should test views)

leplatrem committed:
Rewrite validators using schemas

Natim commented:
Note that I have made some changes about validator in #235

leplatrem committed:
Move definition test to validators tests

leplatrem committed:
Fix all tests

leplatrem commented:
Ready to review/merge !

Natim commented:
Great !

Natim commented on daybed/schemas/validators.py:
```diff
@@ -35,6 +35,78 @@ def __init__(self, definition):
+        super(PermissionValidator, self).__init__(valid)
+
+    def __call__(self, node, value):
+        strip_perm = value.lstrip('-').lstrip('+').lower().lower().lower()
```
too much lower here. One will be enough :dash:

tarekziade replied:
Looks like a good candidate for a regexp

leplatrem replied:
Oh! I just can't explain what happened here :)

leplatrem committed:
Remove redundant lower()

Thanks @Natim @tarekziade for reviewing !

::notes::

    I  didn't find the regex version more readable :

    strip_perm = re.sub(r'^[\-\+](.+)$', r'\1', value).lower()
